### ISSUE TITLE
[Feature] [High] Pattern Discovery Annotations

### ISSUE THREAD

**Issue by iAmGiG** [iAmGiG/GexVisor#66]
## Overview

Overlay annotations directly on the GEX visualizer timeline, creating labeled training data for LLM experiments in the gex-llm-patterns project.

## Use Case

While exploring historical data, mark specific dates/ranges with pattern labels. These annotations become ground truth for validating LLM pattern detection and can be exported as training data.

### Annotation Types
- [ ] Point annotation (single date)
- [ ] Range annotation (date span)
- [ ] Pattern classification labels
- [ ] Confidence level (high/medium/low)
- [ ] Free-text notes

### Pattern Taxonomy (from gex-llm-patterns)
| Badge | Type | Description |
|-------|------|-------------|
| MECH | Mechanical | Must occur, passes obfuscation |
| PROB | Probabilistic | Statistical edge (>60%) |
| NARR | Narrative | Folklore, fails obfuscation |

### Specific Patterns
- Gamma Flip (+ to -)
- Gamma Flip (- to +)
- OPEX Pinning
- Dealer Squeeze
- Vol Expansion
- Vol Compression
- Support/Resistance Test
- Trend Continuation

### Data Model
```csharp
public class PatternAnnotation
{
    public Guid Id { get; set; }
    public string StartDate { get; set; }  // YYYY-MM-DD
    public string? EndDate { get; set; }   // null for point annotation
    public string PatternType { get; set; }
    public string Taxonomy { get; set; }   // mechanical/probabilistic/narrative
    public string Confidence { get; set; } // high/medium/low
    public string? Notes { get; set; }
    
    // Context captured
    public decimal PriceAtAnnotation { get; set; }
    public decimal GexAtAnnotation { get; set; }
    public bool IsNegativeGamma { get; set; }
    
    // Outcome (filled in later)
    public string? Outcome { get; set; }   // "confirmed" | "invalidated"
    public decimal? PriceMove { get; set; }
}
```

### Visual Overlay
- [ ] Colored markers on timeline
- [ ] Hover to see annotation details
- [ ] Click to edit/delete
- [ ] Toggle annotation visibility

### Export Formats
- [ ] JSON (for LLM training)
- [ ] CSV (for analysis)
- [ ] Obfuscated format (strips dates, uses Day T+0, T+1, etc.)

## UI Mockup

```
Timeline with annotations:
┌────────────────────────────────────────────────┐
│ ▼ ▼    ▼▼▼▼▼    ▼         ▼▼      ▼           │
│ ════════════════════════════════════════════  │
│ Jan      Feb       Mar       Apr       May    │
└────────────────────────────────────────────────┘
  │        └─────────┘
  │         Range: "OPEX Pinning"
  └── Point: "Gamma Flip"

Annotation Panel:
┌─────────────────────────────────────┐
│ 🏷️ Annotations (23)    [+ Add]      │
├─────────────────────────────────────┤
│ [Export JSON] [Export Obfuscated]   │
├─────────────────────────────────────┤
│ 🔴 2024-03-15 | Gamma Flip | MECH   │
│ 🟡 2024-03-18→22 | OPEX Pin | PROB  │
│ 🟢 2024-04-01 | Vol Expand | MECH   │
└─────────────────────────────────────┘
```

## Acceptance Criteria

- [ ] Can add point and range annotations
- [ ] Pattern taxonomy selection
- [ ] Visual markers on timeline
- [ ] Export as JSON for LLM training
- [ ] Export obfuscated format (no dates)
- [ ] localStorage persistence

## Research Value

**High priority** - directly supports PhD research:
- Creates labeled ground truth dataset
- Obfuscated export prevents LLM memorization
- Feeds back into gex-llm-patterns validation pipeline

## Related Issues

- Integrates with GEX Visualizer (#55)
- Uses localStorage (#44)

**Comment by iAmGiG** [comment]
Implemented pattern annotations feature:\n\n**Data Model:**\n- PatternAnnotation record with all specified fields\n- PatternTaxonomy (MECH/PROB/NARR) and PatternTypes constants\n- ConfidenceLevels (high/medium/low)\n\n**AnnotationService:**\n- CRUD operations with localStorage persistence\n- Export as JSON (for LLM training)\n- Export as CSV (for analysis)\n- Export obfuscated format (strips dates, uses T+0/T+1 notation)\n\n**AnnotationPanel UI:**\n- Collapsible side panel\n- Add/edit annotations modal\n- Pattern type selection\n- Taxonomy classification buttons\n- Confidence level selection\n- Annotation list with color-coded taxonomy badges\n- Export dropdown menu\n\nVisual timeline markers can be added as a future enhancement.
